lighthouse-core/gather/gather-runner.js

```diff
@@ -215,6 +215,31 @@ class GatherRunner {
     return new LHError(LHError.errors.CHROME_INTERSTITIAL_ERROR);
   }
 
+  /**
+   * Returns an error if we try to load a non-HTML page.
+   * @param {LH.Artifacts.NetworkRequest|undefined} mainRecord
+   * @return {LH.LighthouseError|undefined}
+   */
+  static getDocTypeError(mainRecord) {
+    // MIME types are case-insenstive
+    const HTML_MIME_REGEX = /^text\/html$/i;
+
+    // If we never requested a document, there's no doctype error, let other cases handle it.
+    if (!mainRecord) return undefined;
+
+    // If the main document failed, this error case is undefined, let other cases handle it.
+    if (mainRecord.failed) return undefined;
```

Comment:
probably fine to combine these two cases into a single `if (!mainRecord || mainRecord.failed) return;` since they aren't super important to this check in particular, but could probably just drop the `mainRecord.failed` check completely, since this function isn't so much a check of a valid main document than a check that it's not an invalid media type.

Reply:
Thanks for pointing this out, I ended up just dropping the mainRecord.failed check